### PR TITLE
hatch: init at 1.3.1

### LINKED ISSUE #182726: Packaging request: hatch
**Project description**
Hatch is a modern, extensible Python project manager.

**Metadata**
* homepage URL: https://hatch.pypa.io/latest/
* source URL: https://github.com/pypa/hatch
* license: MIT
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Fixes https://github.com/NixOS/nixpkgs/issues/182726

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
This should probably be next to https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/hatchling/default.nix

**Comment by mweinelt** [comment]
> This should probably be next to [https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/hatchling/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/hatchling/default.nix?rgh-link-date=2022-08-05T18%3A32%3A55Z)

This is the project management CLI, not a library anyone would import, so I disagree. Please make your case.

File: pkgs/development/tools/hatch/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+, python3
+, git
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "hatch";
+  version = "1.3.1";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
```
Comment (ofek) [inline pkgs/development/tools/hatch/default.nix:13]: Please use fetchPypi like https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/hatchling/default.nix

**Review by ofek — CHANGES_REQUESTED** [review]
💯

**Comment by SuperSandro2000** [comment]
> This is the project management CLI, not a library anyone would import, so I disagree. Please make your case.

it is from the same repository and very closely related.
hatcling https://github.com/pypa/hatch/tree/master/backend 
hatch https://github.com/pypa/hatch

File: pkgs/development/tools/hatch/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/hatch/default.nix:13]: That doesn't matter. We often use fetchFromGitHub for various reasons.

**Comment by mweinelt** [comment]
pkgs/tools/package-management/{conda,pdm}

**Comment by ofek** [comment]
new release fyi https://github.com/pypa/hatch/releases/tag/hatch-v1.4.0

**Comment by SuperSandro2000** [comment]
> pkgs/tools/package-management/{conda,pdm}

yeah, true. Well, whatever. Just follow that.

**Pull request merged**